mirrord/vpn/src/config.rs

```diff
@@ -0,0 +1,64 @@
+use ipnet::IpNet;
+use k8s_openapi::api::core::v1::ConfigMap;
+use kube::Api;
+
+#[derive(Debug)]
+pub struct VpnConfig {
+    pub dns_domain: String,
+    pub dns_nameservers: Vec<String>,
+    pub service_subnet: IpNet,
+}
+
+impl VpnConfig {
+    pub async fn from_configmaps(api: &Api<ConfigMap>) -> Option<Self> {
```

Comment:
This function early returns `None` without any error log in multiple places. Maybe you can make it return `anyhow::Result` or a custom error type?